src/accountsdb/db.zig

```diff
@@ -58,18 +59,334 @@ pub const ACCOUNT_INDEX_BINS: usize = 8192;
+    defer accounts_lru.deinit();
+
+    const account = try Account.random(allocator, rng, 1);
+    defer account.deinit(allocator);
+
+    const pubkey = Pubkey.random(rng);
+    const old_slot = 1;
+    const new_slot = 2;
+
+    try accounts_lru.putAccount(old_slot, pubkey, account);
+
+    try std.testing.expectEqual(accounts_lru.get(old_slot, pubkey).?.ref_count.load(.acquire), 1);
+
+    try accounts_lru.copySlot(old_slot, new_slot);
+
+    try std.testing.expectEqual(accounts_lru.get(old_slot, pubkey).?.ref_count.load(.acquire), 2);
+
+    const cached_account = accounts_lru.get(new_slot, pubkey) orelse null;
+    try std.testing.expect(cached_account != null);
+}
+
+// test "AccountsLRU max slots" {
+//     const allocator = std.testing.allocator;
+//     var random = std.rand.DefaultPrng.init(19);
+//     const rng = random.random();
+
+//     var accounts_lru = try AccountsLRU.init(allocator, 10, 2);
+//     defer accounts_lru.deinit();
+
+//     const account = try Account.random(allocator, rng, 1);
+//     defer account.deinit(allocator);
+
+//     const pubkey = Pubkey.random(rng);
+
+//     try accounts_lru.putAccount(1, pubkey, account);
+//     try std.testing.expect(accounts_lru.get(1, pubkey) != null);
+
+//     try accounts_lru.putAccount(2, pubkey, account);
+//     try std.testing.expect(accounts_lru.get(1, pubkey) != null);
+
+//     try accounts_lru.putAccount(3, pubkey, account);
+//     try std.testing.expect(accounts_lru.get(1, pubkey) == null);
+// }
+
+// test "AccountsLRU putAccount returns error on duplicate" {
+//     const allocator = std.testing.allocator;
+//     var random = std.rand.DefaultPrng.init(19);
+//     const rng = random.random();
+
+//     const account = try Account.random(allocator, rng, 1);
+//     defer account.deinit(allocator);
+
+//     var accounts_lru = try AccountsLRU.init(allocator, 10);
+//     defer accounts_lru.deinit();
+
+//     const pubkey = Pubkey.random(rng);
+//     const slot = 1;
+
+//     try accounts_lru.putAccount(slot, pubkey, account);
+
+//     // Trying to insert the same account again should fail
+//     try std.testing.expectEqual(error.InvalidPut, accounts_lru.putAccount(slot, pubkey, account));
+// }
+
+// test "AccountsLRU purgeSlot removes the slot and accounts" {
+//     const allocator = std.testing.allocator;
+//     var random = std.rand.DefaultPrng.init(19);
+//     const rng = random.random();
+
+//     var accounts_lru = try AccountsLRU.init(allocator, 10);
+//     defer accounts_lru.deinit();
+
+//     const account = try Account.random(allocator, rng, 1);
+//     defer account.deinit(allocator);
+
+//     const pubkey = Pubkey.random(rng); // Create a dummy pubkey
+//     const slot = 1;
+
+//     try accounts_lru.putAccount(slot, pubkey, account);
+
+//     // Purge the slot
+//     try accounts_lru.purgeSlot(slot);
+
+//     const result = accounts_lru.get(slot, pubkey);
+//     try std.testing.expect(result == null);
+// }
+
```

Comment:
Should fix and re-enable these tests before merging. There's a memory leak somewhere